model/version.go

```diff
@@ -244,6 +244,22 @@ func (v *Version) MarkFinished(status string, finishTime time.Time) error {
 	)
 }
 
+// UpdateProjectStorageMethod updates the version's parser project storage
+// method.
+func (v *Version) UpdateProjectStorageMethod(method ParserProjectStorageMethod) error {
+	if method == v.ProjectStorageMethod {
```

Comment:
I know this is saving some DB calls but maybe we should consider removing this in case the struct has been modified but the DB has yet to update it?

Reply:
I guess it's possible that this could happen, but I feel like we should usually assume that the in-memory struct hasn't been modified when it's supposed to be updated only after the DB update succeeds. We make this assumption about the struct state in a lot of places.

I know some places in the code base don't follow this struct-DB update state agreement, but they probably should.

Reply:
Fair enough!